src/microbe_stage/editor/MicrobeEditor.cs

```diff
@@ -886,6 +892,47 @@ private void StartMusic()
         Jukebox.Instance.Resume();
     }
 
+    /// <summary>
+    ///   Updates the arrowPosition variable to the top most point of the middle 3 rows
+    ///   Should be called on any layout change
+    /// </summary>
+    private void UpdateArrow()
+    {
+        // 1. Retrieves all the hexes in the microbe
+        // 2. Fetches only the middle 3 rows
+        // 3. Maps the R position to the actual height
+        // 4. Retrieves the minimum value, which is the top most location in the middle 3 rows
+        // The calculation falls back to 0 if there are no hexes found in the middle 3 rows
```

Comment:
This comment is now a bit outdated, as it doesn't literally do this. Also putting the relevant parts next to the lines they are commenting would be very good.